### PR TITLE
feat: support solr client secure option

### PR DESCRIPTION
**Pull request by yoyoyohamapi**
Support solr-client's `secure` options to enable the https connection with solr

### CODE REVIEW

File: templates/admin/plugins/solr.tpl [hunk 1]
```diff
@@ -28,6 +28,10 @@
 						<div class="form-group col-sm-6">
 							<label for="port">Port</label>
 							<input class="form-control" type="text" name="port" id="port" placeholder="8983" />
+						</div>
+							<div class="form-group col-sm-6">
```
Comment (Dravere) [inline templates/admin/plugins/solr.tpl]: Sorry to be pedantic, but the indentation is wrong :smiley:

File: templates/admin/plugins/solr.tpl [hunk 2]
```diff
@@ -28,6 +28,10 @@
...
+							<label for="port">Secure</label>
+							<input class="form-control" type="checkbox" name="secure" id="secure">
```
Comment (julianlam) [inline templates/admin/plugins/solr.tpl]: Recommend you use this style to create a checkbox that looks nicer in Bootstrap: https://github.com/julianlam/nodebb-plugin-markdown/blob/master/public/templates/admin/plugins/markdown.tpl#L10-L15

Also the label should have `for="secure"`

File: library.js [hunk 3]
```diff
@@ -136,6 +137,11 @@ Solr.connect = function () {
 		delete Solr.client;
 	}
 
+	Solr.config = {
+		...Solr.config,
+		secure: Solr.config.secure && Solr.config.secure === 'on'
+	}
+
```
Comment (julianlam) [inline library.js]: Do you need to update `engines` in `package.json` to require a newer node version? Just in case...

Comment (julianlam) [reply library.js]: FYI Node v8.3.0 required for spread operator usage on object literals.

Comment (Dravere) [reply library.js]: FYI there is currently no engine specification in the plugin's `package.json`. But it has a recommendation for NodeBB 1.11 which at the time the requirements were to use NodeJS LTS 8 (which is currently 8.15.1). And currently with NodeBB 1.12 the requirements say to use NodeJS LTS 10.

So not really sure if this is really necessary?

Comment (julianlam) [reply library.js]: Okey dokey :+1: Works for me.

File: templates/admin/plugins/solr.tpl (hunk 2 above)
Comment (yoyoyohamapi) [reply templates/admin/plugins/solr.tpl]: > Recommend you use this style to create a checkbox that looks nicer in Bootstrap: https://github.com/julianlam/nodebb-plugin-markdown/blob/master/public/templates/admin/plugins/markdown.tpl#L10-L15
> 
> Also the label should have `for="secure"`

thx.

**Comment by yoyoyohamapi** [comment]
@julianlam @Dravere fixed style mentioned above. ☺

File: templates/admin/plugins/solr.tpl (hunk 1 above)
Comment (yoyoyohamapi) [reply templates/admin/plugins/solr.tpl]: @fixed

**Pull request merged**